### PR TITLE
chore: refactor codes

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

This pr is to refactor the `StakeHub`, `BSCGovernor` and `TokenRecoverPortal` contracts

### Changes

Notable changes: 
* add `Protectable` contract

### CODE REVIEW

File: contracts/BC_fusion/extension/Protectable.sol [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    function addToBlackList(address account) external virtual onlyProtector {
+        blackList[account] = true;
+        emit BlackListed(account);
+    }
+
+    /**
+     * @dev Remove an address from the black list
+     */
+    function removeFromBlackList(address account) external virtual onlyProtector {
+        blackList[account] = false;
```
Comment (forcodedancing) [inline contracts/BC_fusion/extension/Protectable.sol:87]: How about just delete the key to save storage?

File: contracts/BC_fusion/extension/Protectable.sol [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    function pause() external virtual onlyProtector whenNotPaused {
+        _paused = true;
+        emit Paused();
+    }
+
+    /**
+     * @dev Resume the whole system
+     */
+    function resume() external virtual onlyProtector {
```
Comment (zjubfd) [inline contracts/BC_fusion/extension/Protectable.sol:70]: do you think we need `whenPaused`?

File: contracts/BC_fusion/BSCGovernor.sol [hunk 3]
```diff
@@ -107,40 +77,7 @@ contract BSCGovernor is
...
-        emit BlackListed(account);
-    }
-
-    /**
-     * @dev Remove an address from the black list
-     */
-    function removeFromBlackList(address account) external onlyGovernorProtector {
-        blackList[account] = false;
-        emit UnBlackListed(account);
+        __Protectable_init_unchained(address(0xdEaD));
```
Comment (zjubfd) [inline contracts/BC_fusion/BSCGovernor.sol:80]: Do you need change some script to sed the address here? I guess if any script, it is break now

Comment (Pythonberg1997) [reply contracts/BC_fusion/BSCGovernor.sol:80]: You're right. Fixed

File: contracts/BC_fusion/extension/Protectable.sol (hunk 2 above)
Comment (Pythonberg1997) [reply contracts/BC_fusion/extension/Protectable.sol:70]: Done

File: contracts/BC_fusion/extension/Protectable.sol (hunk 1 above)
Comment (Pythonberg1997) [reply contracts/BC_fusion/extension/Protectable.sol:87]: Done

**Review by zjubfd — APPROVED** [review]

**Review by j75689 — APPROVED** [review]

**Review by forcodedancing — APPROVED** [review]

**Pull request merged**